coreclient/src/groups/mod.rs

```diff
@@ -384,17 +390,16 @@ impl Group {
 
         // Phase 3: Decrypt and verify the infra credentials.
         {
-            let mut connection = pool.acquire().await?;
             for client_auth_info in &client_information {
-                client_auth_info.store(&mut connection).await?;
+                client_auth_info.store(txn).await?;
             }
         }
 
-        let leaf_keys = LeafKeys::load(pool, verifying_key)
+        let leaf_keys = LeafKeys::load(&mut **txn, verifying_key)
             .await?
             .ok_or(anyhow!("Couldn't find matching leaf keys."))?;
         // Delete the leaf signer from the keys store as it now gets persisted as part of the group.
-        LeafKeys::delete(pool, verifying_key).await?;
+        LeafKeys::delete(&mut **txn, verifying_key).await?;
```

Comment:
You should be able to replace `&mut **txn` with txn.as_mut()` everywhere. Not a strong opinion, but `as_mut()` seems a little cleaner to me.